.github/workflows/rust-security-scan.yaml

```diff
@@ -0,0 +1,30 @@
+name: Security & License Scan
```

Comment:
Most of our workflows use the same name here like the respective file name. Could we please rename this to `rust-audit` in both locations. Concise names equally formatted with kebab-case are ealier to digest on the UI and elsewhere.